packages/api/src/router/user.ts

```diff
@@ -32,6 +36,59 @@ export const userRouter = router({
     return user;
   }),
 
+  subscription: privateProcedure.query(async ({ ctx }) => {
+    const stripeCustomer = await ctx.prisma.user
+      .findUnique({
+        where: { id: ctx.user.id },
+      })
+      .stripeCustomer();
+
+    const subscription = await ctx.prisma.stripeSubscription.findFirst({
+      where: {
+        stripeCustomerId: stripeCustomer?.id,
+      },
+      orderBy: {
+        current_period_end: "desc",
+      },
+    });
+
+    if (!subscription) {
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: "No subscription found",
+      });
+    }
+
+    if (
+      StripeSubscriptionStatus.active !== subscription.status &&
+      StripeSubscriptionStatus.trialing !== subscription.status
+    ) {
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: "No active subscription found",
+      });
+    }
+
+    // TODO: Parse differently when other plans are added
+    const membership =
+      subscription.status === StripeSubscriptionStatus.trialing
+        ? Membership.TRIAL
+        : getMembershipFromPriceId(subscription.priceId);
+
+    const period =
+      subscription.priceId === process.env.STRIPE_MONTHLY_PRICE_ID
+        ? "monthly"
+        : "annually";
+
+    return {
+      id: subscription.id,
+      status: subscription.status,
+      current_period_end: subscription.current_period_end,
+      membership,
+      period: period as "monthly" | "annually", // FIXME: Why is `as` needed
```

Comment:
works for me
<img width="605" alt="CleanShot 2022-11-29 at 20 44 02@2x" src="https://user-images.githubusercontent.com/18185649/204632502-3e1cd6ed-3ff5-4c89-8f1b-dd99cc0eca71.png">

Reply:
pushed to see if it works

Reply:
I didn't get the type narrowed down on the frontend without it, so the period would be typed `monthly | annually` on the backend but on the frontend it was just `string`. How's it looking now?

Reply:
That's weird. It's `string`, event though the `period` const is typed accordingly. 🤔

Reply:
reverted